src/domains/bible-books/bible-books.handlers.ts

```diff
@@ -0,0 +1,159 @@
+  }
+}
+
+export async function getBibleBookByBibleIdAndBookId(
+  bibleId: number,
+  bookId: number
+): Promise<Result<BibleBookWithDetails>> {
+  try {
+    const [bibleBook] = await db
+      .select(bibleBookSelect)
+      .from(bible_books)
+      .innerJoin(books, eq(bible_books.bookId, books.id))
+      .innerJoin(bibles, eq(bible_books.bibleId, bibles.id))
+      .where(and(eq(bible_books.bibleId, bibleId), eq(bible_books.bookId, bookId)))
+      .limit(1);
+
+    if (!bibleBook) {
+      return { ok: false, error: { message: 'Bible Book not found' } };
+    }
+
+    return { ok: true, data: bibleBook };
+  } catch {
+    return { ok: false, error: { message: 'Failed to fetch bible book' } };
+  }
+}
+
+export async function createBibleBook(input: CreateBibleBookInput): Promise<Result<BibleBook>> {
+  try {
+    const [bibleBook] = await db.insert(bible_books).values(input).returning();
+
+    if (!bibleBook) {
+      return { ok: false, error: { message: 'Unable to create bible book' } };
+    }
+
+    return { ok: true, data: bibleBook };
+  } catch (error) {
+    if (error instanceof Error && error.message.includes('UNIQUE constraint failed')) {
+      return { ok: false, error: { message: 'Bible book already exists' } };
+    }
+    if (error instanceof Error && error.message.includes('FOREIGN KEY constraint failed')) {
+      return { ok: false, error: { message: 'Invalid bible or book reference' } };
+    }
+
+    return { ok: false, error: { message: 'Failed to create bible book' } };
+  }
+}
+
+export async function updateBibleBook(
+  bibleId: number,
+  bookId: number,
+  input: UpdateBibleBookInput
+): Promise<Result<BibleBook>> {
+  try {
+    const [updated] = await db
+      .update(bible_books)
+      .set(input)
+      .where(and(eq(bible_books.bibleId, bibleId), eq(bible_books.bookId, bookId)))
+      .returning();
+
+    if (!updated) {
+      return { ok: false, error: { message: 'Bible book not found' } };
+    }
+
+    return { ok: true, data: updated };
+  } catch (error) {
+    if (error instanceof Error && error.message.includes('UNIQUE constraint failed')) {
+      return { ok: false, error: { message: 'Bible book already exists with these identifiers' } };
+    }
+    if (error instanceof Error && error.message.includes('FOREIGN KEY constraint failed')) {
+      return { ok: false, error: { message: 'Invalid bible or book reference' } };
+    }
+
+    return { ok: false, error: { message: 'Failed to update bible book' } };
+  }
+}
+
+export async function deleteBibleBook(bibleId: number, bookId: number): Promise<Result<boolean>> {
+  try {
+    const result = await db
+      .delete(bible_books)
+      .where(and(eq(bible_books.bibleId, bibleId), eq(bible_books.bookId, bookId)))
+      .returning({ bibleId: bible_books.bibleId, bookId: bible_books.bookId });
+
+    if (result.length === 0) {
+      return { ok: false, error: { message: 'Bible book not found' } };
+    }
+
+    return { ok: true, data: true };
```

Comment:
We should return the deleted id like we are doing in project handlers.

```suggestion
    return { ok: true, data: { id: result[0].id } };
```